subprojects/platform-jvm/src/main/java/org/gradle/api/java/archives/internal/DefaultAttributes.java

```diff
@@ -48,35 +80,40 @@ public boolean containsValue(Object value) {
 
     @Override
     public Object get(Object key) {
+        if (key instanceof String) {
+            return attributes.get(keyFor((String) key));
+        }
         return attributes.get(key);
     }
 
     @Override
     public Object put(String key, Object value) {
         if (key == null) {
-            throw new ManifestException("The key of a manifest attribute must not be null.");
+            // NPE is the specified exception type (see Map#put javadoc)
```

Comment:
❌ Let's remove the comment here, as noted, we should comply with the `Map` contract.

Reply:
I can remove it, however, I left it for those who would discover the code and ask the classic "WTF?" question. It might prevent back-and-forth changes.